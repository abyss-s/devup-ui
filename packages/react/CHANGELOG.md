# @devup-ui/react

## 0.1.23

### Patch Changes

- 2102fd7: Support theme typing to css

## 0.1.22

### Patch Changes

- e1d8d8b: Fix typing issue

## 0.1.21

### Patch Changes

- 96cd069: Fix theme typing issue

## 0.1.20

### Patch Changes

- 064476f: Fix TypographyKey type definition for better type checking

## 0.1.19

### Patch Changes

- 7e54c1f: Fix type checking issue

## 0.1.18

### Patch Changes

- 77afb9e: Support generic to as prop

## 0.1.17

### Patch Changes

- ea622ed: Add svg props

## 0.1.16

### Patch Changes

- e072762: convert css

## 0.1.15

### Patch Changes

- b40d274: Add typograhpy to props

## 0.1.14

### Patch Changes

- 7f19bd5: Implement ThemeScript, useTheme, getTheme, setTheme

## 0.1.13

### Patch Changes

- 00b14ca: Support optional args in css

## 0.1.12

### Patch Changes

- b84cbc1: Update css.ts

## 0.1.11

### Patch Changes

- 76e8fa1: Add grid, Support for selector in css

## 0.1.10

### Patch Changes

- 0da9456: Add content to style props

## 0.1.9

### Patch Changes

- 6f7c55d: Add group selector

## 0.1.8

### Patch Changes

- 8943bad: Add double selector

## 0.1.7

### Patch Changes

- 0040445: Fix selector type

## 0.1.6

### Patch Changes

- b25c769: Add placehoder to selector

## 0.1.5

### Patch Changes

- e9d655f: Add shorthand

## 0.1.4

### Patch Changes

- 7bde1f2: Add boxSizing prop

## 0.1.3

### Patch Changes

- 30443ae: Update readme

## 0.1.2

### Patch Changes

- 242a6c2: Update package

## 0.1.1

### Patch Changes

- 6100644: Add Image component

## 0.1.0

### Minor Changes

- c0ff96f: Deploy
