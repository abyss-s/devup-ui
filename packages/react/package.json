{
  "name": "@devup-ui/react",
  "description": "Zero Config, Zero FOUC, Zero Runtime, CSS in JS Preprocessor",
  "repository": "https://github.com/dev-five-git/devup-ui",
  "author": "devfive",
  "license": "Apache-2.0",
  "homepage": "https://devup-ui.com",
  "bugs": {
    "url": "https://github.com/dev-five-git/devup-ui/issues",
    "email": "contact@devfive.kr"
  },
  "keywords": [
    "css",
    "css-in-js",
    "css-in-js-preprocessor",
    "css-in-js-framework",
    "react"
  ],
  "version": "1.0.1",
  "type": "module",
  "scripts": {
    "lint": "eslint",
    "build": "tsc && vite build"
  },
  "publishConfig": {
    "access": "public"
  },
  "sideEffects": false,
  "main": "./dist/index.cjs",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.cjs"
    }
  },
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "dependencies": {
    "react": "^19.1",
    "csstype": "^3.1"
  },
  "devDependencies": {
    "rollup-plugin-preserve-directives": "^0.4.0",
    "vite": "^6.3.5",
    "vite-plugin-dts": "^4.5.4",
    "vitest": "^3.1.3",
    "typescript": "^5.8.3",
    "@types/react": "^19"
  },
  "peerDependencies": {
    "react": "*",
    "csstype": "*"
  }
}