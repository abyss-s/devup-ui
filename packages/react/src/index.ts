export { Box } from './components/Box'
export { Button } from './components/Button'
export { Center } from './components/Center'
export { Flex } from './components/Flex'
export { Grid } from './components/Grid'
export { Image } from './components/Image'
export { Input } from './components/Input'
export { Text } from './components/Text'
export { ThemeScript } from './components/ThemeScript'
export { VStack } from './components/VStack'
export { useTheme } from './hooks/use-theme'
export type { DevupProps } from './types/props'
export type { DevupTheme, DevupThemeColors } from './types/theme'
export type { DevupThemeTypography } from './types/typography'
export { css } from './utils/css'
export { getTheme } from './utils/get-theme'
export { globalCss } from './utils/global-css'
export { initTheme } from './utils/init-theme'
export { setTheme } from './utils/set-theme'
