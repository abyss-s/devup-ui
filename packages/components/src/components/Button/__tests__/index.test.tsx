import { render } from '@testing-library/react'
import { expect } from 'vitest'

import { Button } from '../index'

describe('Button', () => {
  it('should render', () => {
    const { container } = render(<Button>Click me</Button>)
    expect(container).toMatchSnapshot()
  })

  it('should disable', () => {
    const { container } = render(<Button disabled>Click me</Button>)
    expect(container).toMatchSnapshot()
    expect(container.querySelector('button')).toHaveAttribute('disabled')
  })
})
