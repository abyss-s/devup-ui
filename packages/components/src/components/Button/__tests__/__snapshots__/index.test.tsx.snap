// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should disable 1`] = `
<div>
  <button
    aria-disabled="true"
    aria-label="button"
    class="d16 d17 d18 d19 d20 d21 d1 d2 d3 d22 d23 typo-buttonxs d24 d25 d26 d27 d28 d7 d8 d9 d10 d69 d70 d71 d72 d73 d74 d75 d76 d77 d78 d79 d80 d81 d82 d83 d84 d85 d86  "
    disabled=""
    type="button"
  >
    <div
      class="d65 d66 d67 d68"
    >
      <div
        class=" d62 d63 "
      >
        Click me
      </div>
    </div>
  </button>
</div>
`;

exports[`Button > should render 1`] = `
<div>
  <button
    aria-label="button"
    class="d16 d17 d18 d19 d20 d21 d1 d2 d3 d22 d23 typo-buttonxs d24 d25 d26 d27 d28 d7 d8 d9 d10 d69 d70 d71 d72 d73 d74 d75 d76 d77 d78 d79 d80 d81 d82 d83 d84 d85 d86  "
    type="button"
  >
    <div
      class="d65 d66 d67 d68"
    >
      <div
        class=" d62 d63 "
      >
        Click me
      </div>
    </div>
  </button>
</div>
`;
